app/forms/portal/sign_form8879.rb

```diff
@@ -0,0 +1,69 @@
+module Portal
+  class SignForm8879 < Form
+    attr_accessor :primary_accepts_terms, :primary_confirms_identity, :ip, :transaction_failed
+    validate :terms_accepted
+    validate :identity_confirmed
+
+    def initialize(tax_return, params = {})
+      @tax_return = tax_return
+      super(params)
+    end
+
+    def sign
+      return false unless valid?
```

Comment:
just to confirm, we are checking that it is valid once in the controller and once here? i think we can probably skip this check unless it's the case in our other forms that we check validity at the beginning of what is usually the `save` method